kentikapi/internal/synthetics/test.go

```
@@ -0,0 +1,652 @@
+func testFromPayload(t *syntheticspb.Test) (*synthetics.Test, error) {
+	if t == nil {
+		return nil, fmt.Errorf("response payload is nil")
+	}
+
+	if t.Id == "" {
+		return nil, fmt.Errorf("empty test ID in response payload")
+	}
+
+	settings, err := testSettingsFromPayload(t)
+	if err != nil {
+		return nil, fmt.Errorf("convert response payload to test settings: %w", err)
+	}
+
+	createdBy := userInfoPtrFromPayload(t.CreatedBy)
+	if createdBy == nil {
+		return nil, errors.New("createdBy field is nil in response payload")
+	}
+
+	return &synthetics.Test{
+		Name:          t.Name,
+		Status:        synthetics.TestStatus(t.Status.String()),
+		Settings:      settings,
+		ID:            t.Id,
+		Type:          synthetics.TestType(t.Type),
+		CreateDate:    t.Cdate.AsTime(),
+		UpdateDate:    t.Edate.AsTime(),
+		CreatedBy:     *createdBy,
+		LastUpdatedBy: userInfoPtrFromPayload(t.LastUpdatedBy),
+	}, nil
+}
+
+func testSettingsFromPayload(t *syntheticspb.Test) (synthetics.TestSettings, error) {
+	if t.Settings == nil {
+		return synthetics.TestSettings{}, fmt.Errorf("empty test settings")
+	}
+	ts := t.Settings
+
+	definition, err := testDefinitionFromPayload(t)
+	if err != nil {
+		return synthetics.TestSettings{}, err
+	}
+
+	hs, err := healthSettingsFromPayload(ts.HealthSettings)
+	if err != nil {
+		return synthetics.TestSettings{}, err
+	}
+
+	return synthetics.TestSettings{
+		Definition:           definition,
+		AgentIDs:             ts.AgentIds,
+		Tasks:                taskTypesFromPayload(ts.Tasks),
+		Health:               hs,
+		Ping:                 pingSettingsFromPayload(ts.Ping),
+		Traceroute:           tracerouteSettingsFromPayload(ts.Trace),
+		Period:               time.Duration(ts.Period) * time.Second,
+		Family:               synthetics.IPFamily(ts.Family.String()),
+		NotificationChannels: ts.NotificationChannels,
+	}, nil
+}
+
+// nolint: gocyclo
+func testDefinitionFromPayload(t *syntheticspb.Test) (synthetics.TestDefinition, error) {
```

There is also `bgp_monitor` test type which should be silently ignored (it will be handled in separate BGP monitoring API).

---

The API may already return `transaction` type test, but it cannot yet provide actual configuration or results.

---

Added code for bgp_monitor tests: 4e80fa9

---

Added code for transaction tests: 3d7e8d2